packages/marshal/src/stringify-capdata.js

```diff
@@ -0,0 +1,57 @@
+/** @template S @typedef {import('./types.js').CapData<S>} CapData */
+
+const { Fail, quote: q } = assert;
+
+/** @typedef {string} SimpleString */
+
+/**
+ * A SimpleString is one where JSON.strigify-ing the string just
+ * puts quotes around the contents, and it contains no close square
+ * brackets. Many slot encodings use slot strings that
+ * obey these rules, which is why they are useful.
+ *
+ * @param {unknown} str
+ * @returns {asserts str is SimpleString}
+ */
+export const assertSimpleString = str => {
+  assert(typeof str === 'string');
+  const stringified = JSON.stringify(str);
+  const expected = `"${str}"`;
+  stringified === expected ||
+    Fail`Expected to stringify to ${q(expected)}, not ${q(stringified)}}`;
+  assert(str.indexOf(']') === -1);
+};
+harden(assertSimpleString);
+
+/**
+ * @param {CapData<SimpleString>} capData
+ * @returns {string}
+ */
+export const stringifyCapData = capData => {
+  const {
+    body,
+    slots: [...slots],
+  } = capData;
+  assert(typeof body === 'string');
+  assert(body.startsWith('#'));
+  const scBody = body.slice(1);
+  JSON.parse(scBody); // Just asserts that it parses as JSON
+  slots.forEach(assertSimpleString);
+  return `{"slots":${JSON.stringify(slots)},"#body":${scBody}}`;
+};
+harden(stringifyCapData);
+
+const DSL = /^\{"slots":(\[[^\]]*\]),"#body":(.*)\}$/;
```

Comment:
I would recommend to move to a less greedy search and string boundaries if possible
`\A\{"slots":(\[[^\]]+?\]),"#body":(.+?)\}\Z`

Reply:
I had never seen `\A` or `\Z` before. Searching for it in JS RegExp documentation finally found it at https://docstore.mik.ua/orelly/webprog/jscript/ch10_01.htm in a section titled 
"Perl RegExp Features Not Supported in JavaScript"

But good point about newlines. Where the `(.*)` appears, we should accept newlines along with everything else. IIUC, adding the `s` flag (aka "dotAll") on the end will give "." that meaning. Of course there are other ways to get the same effect by changing the pattern rather than adding a flag. What do you recommend?